fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/BaseProtocolMetricMaker.java

```diff
@@ -59,12 +61,68 @@ public LogicalMetric makeInnerWithResolvedDependencies(
+        LogicalMetric dependentMetric = dependentMetrics.get(0);
+        LogicalMetric renamedDependentMetric = renameIfConflicting(logicalMetricInfo.getName(), dependentMetric);
+        List<LogicalMetric> renamedDependentMetrics =
+                dependentMetrics.size() == 1 ? Collections.singletonList(renamedDependentMetric) : dependentMetrics;
+        TemplateDruidQuery partialQuery = makePartialQuery(logicalMetricInfo, renamedDependentMetrics);
+        ResultSetMapper calculation = makeCalculation(logicalMetricInfo, renamedDependentMetrics);
+        ProtocolSupport protocolSupport = makeProtocolSupport(logicalMetricInfo, renamedDependentMetrics);
         return new ProtocolMetricImpl(logicalMetricInfo, partialQuery, calculation, protocolSupport);
     }
 
+    /**
+     * Renames the provided logical metric if there is a name conflict between it and the final desired output name.
+     * If there is no name conflict the input metric is returned unchanged. Otherwise, the input metric is renamed to
+     * not conflict with the final output name.
+     * <p>
+     * This method specifically the {@link GeneratedMetricInfo} typing on the input metric if it has to be renamed. All
+     * other subclasses of {@link LogicalMetricInfo} are lost during the renaming. If maintaining more LogicalMetricInfo
+     * types is required, a withName method must «should be added to the LogicalMetricInfo interface.
+     *
+     * @param finalOutputName  The name to check for conflicts on
+     * @param dependentMetric  The dependent metric to check for a conflicting name
+     * @return  The metric without a name conflicting with finalOutputName
+     */
+     protected LogicalMetric renameIfConflicting(String finalOutputName, LogicalMetric dependentMetric) {
+        LogicalMetricInfo dependentMetricInfo = dependentMetric.getLogicalMetricInfo();
+        //if no name conflict ,  return the original dependentMetric
+        if (!java.util.Objects.equals(finalOutputName, dependentMetricInfo.getName())) {
+            return dependentMetric;
+        }
+        String newName = getRenamedMetricNameWithPrefix(dependentMetricInfo.getName());
+        while (ifConflicting(newName, dependentMetric)) {
+            newName = getRenamedMetricNameWithPrefix(dependentMetricInfo.getName());
+        }
+        LogicalMetricInfo resultInfo;
+        if (dependentMetricInfo instanceof GeneratedMetricInfo) {
+            GeneratedMetricInfo generatedMetricInfo = (GeneratedMetricInfo) dependentMetricInfo;
+            resultInfo = new GeneratedMetricInfo(newName, generatedMetricInfo.getBaseMetricName());
+        } else {
+            resultInfo = new LogicalMetricInfo(newName);
+        }
+
+        return dependentMetric.withLogicalMetricInfo(resultInfo);
+    }
+
+    /**
+     * Abstract method to build new aggregation name.
+     * @param name The dependent metric name need rename.
+     * @return Renamed metric name with specified prefix
+     */
```

Comment:
empty line between name and param and param and return.

Reply:
javadoc shouldn't call out that it's abstract, rather, describe the expected contract, so that javadoc by default describes the contract of implementing subclasses.